src/flb_input_chunk.c

```diff
@@ -632,3 +661,41 @@ int flb_input_chunk_get_tag(struct flb_input_chunk *ic,
     return ret;
 
 }
+
+/*
+ * Iterate all chunks associated to the input instance and summarize
+ * the total number of bytes in use to all related output instances.
+ */
+void flb_input_chunk_update_output_instances(struct flb_input_chunk *ic)
+{
+    ssize_t bytes;
+    uint64_t routes_mask;
+    uint64_t o_mask_id;
+    struct mk_list *head;
+    struct mk_list *tmp;
+    struct flb_output_instance *o_ins;
+
+    if (cio_chunk_is_up(ic->chunk) == CIO_FALSE) {
+        return;
+    }
+
+    bytes = flb_input_chunk_get_size(ic);
+    if (bytes <= 0) {
+        return;
+    }
+
+    /* for each output plugin, we update the fs_chunks_size */
+    mk_list_foreach_safe(head, tmp, &ic->in->config->outputs) {
+        o_ins = mk_list_entry(head, struct flb_output_instance, _head);
+        o_mask_id = o_ins->mask_id;
+        routes_mask = ic->routes_mask;
```

Comment:
ic mask never changes, variable not needed.

Reply:
updated